src/Microsoft.OData.Core/JsonLight/ODataJsonLightContextUriParser.cs

```diff
@@ -71,19 +71,44 @@ private ODataJsonLightContextUriParser(IEdmModel model, Uri contextUriFromPayloa
             ODataPayloadKind payloadKind,
             Func<IEdmType, string, IEdmType> clientCustomTypeResolver,
             bool needParseFragment,
-            bool throwIfMetadataConflict = true)
+            bool throwIfMetadataConflict = true,
+            Uri baseUri = null,
+            string navigationSource = null)

```

Comment:
Why not make this `IEdmNavigationSource navigationSource` instead of `string navigationSource`. You'd avoid doing `this.CurrentNavigationSource?.Name` is so many place, plus I believe it's the better option

Reply:
fixed